src/lib.rs

```diff
@@ -101,12 +102,12 @@ pub fn reflink_or_copy(from: impl AsRef<Path>, to: impl AsRef<Path>) -> io::Resu
 }
 
 fn check_is_file_and_error(from: &Path, err: io::Error) -> io::Error {
-    if from.is_file() {
+    if fs::symlink_metadata(from).map_or(false, |m| m.is_file() || m.is_dir()) {
```

Comment:
Since only macos support reflink dir, we should only check for `is_dir` on MacOS/iOS

```suggestion
    if fs::symlink_metadata(from).map_or(false, |m| m.is_file() || (cfg!(any(target_os = "macos", target_os = "ios")) && m.is_dir())) {
```

Reply:
Also it seems like that on MacOS clonefile also supports symlink according to https://www.manpagez.com/man/2/clonefile/